lib/elixir/src/elixir_def.erl

```diff
@@ -172,7 +172,7 @@ def_to_clauses(Kind, Meta, _Args, _Guards, nil, E) ->
 def_to_clauses(_Kind, Meta, Args, Guards, [{do, Body}], _E) ->
   [{Meta, Args, Guards, Body}];
 def_to_clauses(_Kind, Meta, Args, Guards, Body, _E) ->
-  [{Meta, Args, Guards, {'try', Meta, [Body]}}].
+  [{Meta, Args, Guards, {'try', [{origin,  def} | Meta], [Body]}}].
```

Comment:
Is this OK? I could not think in another way of passing this info forward.

Reply:
This looks good but you want to pass `Kind` instead of `def`. :)

Reply:
Nice catch! What about `origin` as the key?

Reply:
Looks fine. :)